src/Wt/WTableRow.C

```diff
@@ -87,6 +88,7 @@ WTableCell *WTableRow::elementAt(int column)
 int WTableRow::rowNum() const
 {
   if (table_)
+    if (rowNumber_ != -1) return rowNumber_;
     for (unsigned i = 0; i < table_->rows_.size(); ++i)
       if (table_->rows_[i].get() == this)
 	return i;
```

Comment:
In what scenario would it have to fallback to this code?

Also: this loop is not guarded by the above `if (table_)` anymore. The [readbility-misleading-indentation](https://clang.llvm.org/extra/clang-tidy/checks/readability/misleading-indentation.html) clang-tidy check catches this.

It seems to me the entire function could be simplified to `return rowNumber_;`.